lib-multisrc/grouple/src/eu/kanade/tachiyomi/multisrc/grouple/GroupLe.kt

```diff
@@ -313,20 +306,22 @@ abstract class GroupLe(
 
         val html = document.html()
 
-        val readerMark = "rm_h.readerDoInit(["
+        if (document.select(".user-avatar").isEmpty() &&
+            document.title().run { contains("AllHentai") || contains("MintManga") || contains("МинтМанга") }
 
-        if (!html.contains(readerMark)) {
-            if (document.select(".input-lg").isNotEmpty() || (
-                document.select(".user-avatar")
-                    .isEmpty() && document.select("img.logo").first()?.attr("title")
-                    ?.contains("Allhentai") == true
-                )
-            ) {
-                throw Exception("Для просмотра контента необходима авторизация через WebView\uD83C\uDF0E")
-            }
-            if (!response.request.url.toString().contains(baseUrl)) {
+        ) {
+            throw Exception("Для просмотра контента необходима авторизация через WebView\uD83C\uDF0E")
+        }
+
+        val readerMark = when {
+            html.contains("rm_h.readerDoInit([") -> "rm_h.readerDoInit(["
+            html.contains("rm_h.readerInit([") -> "rm_h.readerInit(["
+            !response.request.url.toString().contains(baseUrl) -> {
                 throw Exception("Не удалось загрузить главу. Url: ${response.request.url}")
             }
+            else -> {
+                throw Exception("Дизайн сайта обновлен, для дальнейшей работы необходимо обновление дополнения")
+            }
```

Comment:
[maybe this?](https://github.com/KotatsuApp/kotatsu-parsers/commit/04225170d34e486be3925cdb517d32b62c404b6d#diff-b47c4e01a1271f7078a22f4d0a98e7747abf2e163a825c11c9b9852ae86155b4R214-R226)

Reply:
It used to check for another "readerMark" variant, but I couldn’t find it in any current grouple extensions. The way the page parsing works now covers all existing grouple extensions. Or did I misunderstand what you was pointing out?

Reply:
Maybe he meant using
`data.indexOf("rm_h.readerInit(") `
instead of 
`html.contains("rm_h.readerInit([")`

Reply:
the commit I linked uses three different signatures, `rm_h.readerDoInit(`, `rm_h.readerInit( 0,`, `.readerInit(`, so I was wondering if those should be used to check these difference between source versions  (as pointed out by Exinaus 
> theoretically, AllHentai uses different (previous) function because work on this site is still in progress

)
becuase I don't think we should be checking for source title in the multisrc code, or am I misunderstanding something?

Reply:
That's the name of JavaScript function where site stores all the images for a chapter
![Безымянный](https://github.com/user-attachments/assets/c02605fc-c658-44ad-8124-9b8b086e4b45)

Where else it should be checked?